[Question] "Caused by: java.lang.IllegalStateException: Only KClass supported as classifier, got T"

Could you please teach me how to fix the the exception on the following code?
***Caused by: java.lang.IllegalStateException: Only KClass supported as classifier, got T***

My code is the following.

```
@Serializable
data class Params<T>(val params: T)        

val rparams7 = listOf(mapOf("settings" to listOf(mapOf("value" to "2", "target" to "color"))))
val params7  = Params(rparams7)
Log.d(TAG,"params7: ${Json.encodeToString(params7)}") // → params7: {"params":[{"settings":[{"value":"2","target":"color"}]}]}
```

This Serialization and Json.encodeToString works with no problem.

Next, I prepare the xpost func then pass the rparams7 as the raw data to that function which execute the same process
Serialization and Json.encodeToString likely the above code.

```
val rparams7 = listOf(mapOf("settings" to listOf(mapOf("value" to "2", "target" to "color"))))
xpost(rparams7)

fun <T> xpost(params: List<T>) {
    val x = Params(params)
    Log.d(TAG,"xpost x : $x") // -> xpost x : Params(params=[{settings=[{value=2, target=color}]}])
    val y = Json.encodeToString(x)　 // Caused by: java.lang.IllegalStateException: Only KClass supported as classifier, got T
    Log.d(TAG,"xpost Json.encodeToString : $y")
    // myhttp.post(url, y)
}
```

But xpost Json.encodeToString has the Exception.
***Caused by: java.lang.IllegalStateException: Only KClass supported as classifier, got T***

Could you advice me how to fix this exception.
I just want to pass the Json.encodeToString value from xpost function to the onward function.


**Environment**
Android Studio Bumblebee 2021.1.1
build.gradle (Project: My_Application) classpath "org.jetbrains.kotlin:kotlin-serialization:1.6.10"
build.gradle (Module: My_Application.app) implementation 'org.jetbrains.kotlinx:kotlinx-serialization-json:1.3.2'

--- a/core/commonMain/src/kotlinx/serialization/internal/Platform.common.kt
+++ b/core/commonMain/src/kotlinx/serialization/internal/Platform.common.kt
@@ -99,6 +99,11 @@ internal expect fun KClass<*>.platformSpecificSerializerNotRegistered(): Nothing
 @Suppress("UNCHECKED_CAST")
 internal fun KType.kclass() = when (val t = classifier) {
     is KClass<*> -> t
+    is KTypeParameter -> {
+        error("Captured type paramerer $t from generic non-reified function. " +
+                "Such functionality cannot be supported as $t is erased, either specify serializer explicitly or make " +
+                "calling function inline with reified $t")
+    }
     else -> error("Only KClass supported as classifier, got $t")
 } as KClass<Any>
 

